presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -64,6 +64,20 @@ public DictionaryBlock(int positionCount, Block dictionary, int[] ids, boolean d
         this(0, positionCount, dictionary, ids, dictionaryIsCompacted, dictionarySourceId);
     }
 
+    public static Block mask(Block sourceBlock, int[] ids)
```

Comment:
FWIW, `master` now has `com.facebook.presto.spi.Page#mask(Page, int[])`. It could be refactored to call this method for every block.